bundle-httpcache/src/main/java/com/adobe/acs/commons/httpcache/config/impl/HttpCacheConfigImpl.java

```diff
@@ -0,0 +1,131 @@
+package com.adobe.acs.commons.httpcache.config.impl;
+
+import com.adobe.acs.commons.httpcache.config.HttpCacheConfig;
+import com.adobe.acs.commons.httpcache.store.HttpCacheStore;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.felix.scr.annotations.*;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Config class for http cache. Modelled as OSGi config factory. Every factory instance is tied to a request URI.
+ * Parameters such as selection of cache store, user groups, invalidation details shall also be configured.
+ */
+@Component(label = "ACS AEM Commons - HTTP Cache - Cache config",
+           description = "Config for request URI pattern that has to be cached. Each config is tied to a single " +
+                   "request URI pattern.",
+           configurationFactory = true,
+           metatype = true)
+@Service
+public class HttpCacheConfigImpl implements HttpCacheConfig {
+    private static final Logger log = LoggerFactory.getLogger(HttpCacheConfigImpl.class);
+
+    @Property(label = "Request URI pattern",
+              description = "Request URI pattern (REGEX) to be cached. Example - /content/mysite(.*).product-data" +
+                      ".json. Mandatory parameter.")
+    private static final String PROP_REQUEST_URI_PATTERN = "httpcache.config.requesturi.pattern";
+    private String requestUriPattern;
+
+    @Property(label = "Is request authenticated",
+              description = "Select if the request is authenticated. If not selected, this cache will be applicable "
+                      + "only for anonymous (public user) requests.",
+              boolValue = HttpCacheConfigImpl.DEFAULT_IS_REQUEST_AUTHENTICATION_REQUIRED)
+    private static final String PROP_IS_REQUEST_AUTHENTICATION_REQUIRED = "httpcache.config.request.authentication" +
+            ".isrequired";
+    private static final boolean DEFAULT_IS_REQUEST_AUTHENTICATION_REQUIRED = false;
+    private boolean isRequestAuthenticationRequired;
+
+    @Property(label = "AEM user groups",
```

Comment:
Am thinking it might be better to have the user define a "HttpCacheConfigKeyGenerator" ... That way custom cache key implementations should be created.

```
public interface HttpCacheConfigKeyGenerator {
    public String getKey(SlingHttpServletRequest request);
}
```

and our impl could be something like ...

```
public class GroupBasedKeyGenerator implements HttpCacheConfigKeyGenerator {
    public String getKey(SlingHttpServletRequest request) {
          // Psuedo-code
          String path = request.getPathInfo();
          String groups = request.adaptTo(Autorizable.class).getGroups().join("_");
          return path + "-" + groups; 
    }
}
```

We could specific the component.pid to get the service reference ..